### PR TITLE
Authentication method tests

### PR DESCRIPTION
**Pull request by lloydwatkin**
Tests for __PLAIN__, __DIGEST-MD5__, and __ANONYMOUS__ authentication methods in integration and browser based tests.

Note: Skipped __DISGEST-MD5__ test in the browser based BOSH test. This issue is documented in #15.

### CODE REVIEW

File: test/resources/prosody.cfg.lua [hunk 1]
```diff
@@ -46,3 +46,11 @@ VirtualHost "localhost"
 
 Component "component.localhost"
 	component_secret = "mysecretcomponentpassword"
+
+VirtualHost "anon.localhost"
+   authentication = "anonymous"
+	enabled = true
+    ssl = {
```
Comment (linkmauve) [inline test/resources/prosody.cfg.lua]: This indentation should be a single tab.

Comment (sonnyp) [inline test/resources/prosody.cfg.lua]: we use spaces actually

Comment (linkmauve) [inline test/resources/prosody.cfg.lua]: Prosody doesn’t.

Comment (sonnyp) [inline test/resources/prosody.cfg.lua]: does it matters for prosody.cfg.lua?

Comment (linkmauve) [inline test/resources/prosody.cfg.lua]: Indentation doesn’t matter at all in Lua, but being consistent is always nice, and people may want to copy/paste part of this configuration into theirs, for example.

Comment (sonnyp) [inline test/resources/prosody.cfg.lua]: @linkmauve true, I happened to do that exactly :)

Comment (lloydwatkin) [inline test/resources/prosody.cfg.lua]: Its fine, I test using this locally (and the travis tests pass too)

**Pull request merged**